var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -561,26 +587,33 @@ def common_config_args(self):
             self.prefix,
             "-v",
             "-opensource",
-            "-{0}opengl".format("" if "+opengl" in spec else "no-"),
             "-{0}".format("debug" if "+debug" in spec else "release"),
             "-confirm-license",
             "-optimized-qmake",
             "-no-pch",
         ]
 
+        # Windows currently only supports the desktop provider for opengl
+        if "+opengl" in spec:
+            config_args.append("-opengl")
+            if IS_WINDOWS:
+                config_args.append("desktop")
```

Comment:
(question) Is it essential to set this and the `QT_OPENGL` env var?

Reply:
One sets it for the active Qt build to ensure Qt build the correct GL widget/plugin and only that one, and the other sets it for dependent packages so Qt knows which widget to bring in as a dependency, or so Qt can error more appropriately (in the case where a dependent project explicitly requests a different backend)